heron/spi/src/java/com/twitter/heron/spi/utils/LauncherUtils.java

```diff
@@ -0,0 +1,185 @@
+  /**
+   * Builds initial runtime config instance using topology information.
+   *
+   * @return initial runtime config instance
+   */
+  public static Config getPrimaryRuntime(TopologyAPI.Topology topology,
+                                         SchedulerStateManagerAdaptor adaptor) {
+    return Config.newBuilder()
+        .put(Keys.topologyId(), topology.getId())
+        .put(Keys.topologyName(), topology.getName())
+        .put(Keys.topologyDefinition(), topology)
+        .put(Keys.schedulerStateManagerAdaptor(), adaptor)
+        .put(Keys.numContainers(), 1 + TopologyUtils.getNumContainers(topology)).build();
+  }
+
+  private static Config putPackingInRuntimeConfig(Config runtime, PackingPlan packing) {
+    Config ytruntime;
+    if (packing == null) {
+      ytruntime = runtime;
+    } else {
+      ytruntime = Config.newBuilder()
+          .putAll(runtime)
+          .put(Keys.instanceDistribution(), packing.getInstanceDistribution())
+          .put(Keys.componentRamMap(), packing.getComponentRamDistribution())
+          .put(Keys.numContainers(), 1 + packing.getContainers().size())
+          .build();
+    }
+    return ytruntime;
+  }
+
+  private static LauncherUtils getInstance() {
+    return instance;
+  }
+
+  /**
+   * For testing only. Could be used to setup a mock
+   */
+  public static void setInstance(LauncherUtils instance) {
+    LauncherUtils.instance = instance;
+  }
+
+  /**
+   * Creates and initializes scheduler instance
+   *
+   * @return initialized scheduler instances
+   */
+  private IScheduler _getSchedulerInstance(
+      Config config,
+      Config runtime,
+      PackingPlan packing) {
+    String schedulerClass = Context.schedulerClass(config);
+    IScheduler scheduler;
+    try {
+      // create an instance of scheduler
+      scheduler = ReflectionUtils.newInstance(schedulerClass);
+    } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
+      LOG.log(Level.SEVERE, "Failed to instantiate scheduler", e);
+      return null;
+    }
+
+    Config ytruntime = putPackingInRuntimeConfig(runtime, packing);
+
+    scheduler.initialize(config, ytruntime);
+    return scheduler;
+  }
+
+  /**
+   * This method can be mocked for unit testing.
+   */
+  public PackingPlan _createPackingPlan(final Config config, final Config runtime) {
+    // Create an instance of the packing class
+    String packingClass = Context.packingClass(config);
+    IPacking packing;
+    try {
+      // create an instance of the packing class
+      packing = ReflectionUtils.newInstance(packingClass);
+    } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
+      LOG.log(Level.SEVERE, "Failed to instantiate packing instance", e);
+      return null;
+    }
+
+    try {
+      packing.initialize(config, runtime);
+      PackingPlan packedPlan = packing.pack();
+      return packedPlan;
+    } finally {
+      SysUtils.closeIgnoringExceptions(packing);
```

Comment:
This is a side effect hidden to the caller. Could you document this effect in the relevant public javadocs.